### ISSUE TITLE
Can one set up automated config file with a password prompt?

### ISSUE THREAD

**Issue by evanbiederstedt** [jupyter/notebook#1700]
Configuration files are output by the command 

```
jupyter notebook --generate-config
```

and placed in some location `~/.jupyter/jupyter_notebook_config.py`. 

In order to set a password, you must begin python and run 

```
from notebook.auth import passwd
passwd()
```

then copy the hashed password, and place it into the config file at

```
c.NotebookApp.password = u''
```

Is it possible to simply write an automatic script that prompts users to enter a password, and then puts this hashed password into the `config` file?

**Comment by paderijk** [comment]
I wrote a [small python script](https://github.com/paderijk/jupyter-password) that creates the correct password entry. Although would be nice if there is a flag like `--set-password`, so for example:

```bash
jupyter notebook --generate-config --set-password
```

Or to reset it:
```bash
jupyter notebook --set-password
```

**Comment by takluyver** [comment]
#2007 added `jupyter notebook password`. This should be released in 5.0.
